script/concerns/fix_ip_module.rb

```diff
@@ -0,0 +1,56 @@
+module FixIpModule
+  # look for the data center host record
+  # returns line from traceroute or nil
+  def get_matching_traceroute(ip:, reg:)
+    traceroute =  `traceroute -m 20 #{ip}`.split("\n")
+    last_ovh_ip = nil
+    traceroute.each do |line|
+      next unless line.match?(reg)
+      # furthest ovh ip from traceroute
+      last_ovh_ip = line
+    end
+    last_ovh_ip
+  end
+
+  # update or create new IpOverride with host data
+  def setup_ip_override(ip:, host_data:, host:)
+    ipor = IpOverride.find_or_create_by(address: ip.address)
+    ipor.traits_autonomous_system_number = ip.traits_autonomous_system_number
+    ipor.country_iso_code = host_data[:country_iso_code]
+    ipor.country_name = host_data[:country_name]
+    ipor.city_name = host_data[:city_name]
+    ipor.data_center_key = host_data[:data_center_key]
+    ipor.data_center_host = host
+    ipor.save
+  end
+
+  def update_ip_with_overrides(aso:)
+    sql1 = "
+      UPDATE ips ip
+      INNER JOIN ip_overrides ipor
+      ON ip.address = ipor.address
+      SET
+      ip.traits_autonomous_system_number = ipor.traits_autonomous_system_number,
+      ip.country_iso_code = ipor.country_iso_code,
+      ip.country_name = ipor.country_name,
+      ip.city_name = ipor.city_name,
+      ip.data_center_key = ipor.data_center_key,
+      ip.data_center_host = ipor.data_center_host,
+      ip.traits_autonomous_system_organization = '#{aso}',
```

Comment:
This is a serious logic error that got carried through from the original Hetzner script. The origin of the error is my mistake.

This query is updating ALL ips with the data from a matching ip_override AND we are setting the ASO for all of the matching ips. Obviously, we only want to update the `ip.traits_autonomous_system_organization` with the correct ASO. I will make some changes on master branch and then we can merge master into this branch to get everything working correctly.

Reply:
Actually, I'll fix it on this branch and we will quickly merge this into master.